controllers/drplacementcontrol_controller.go

```diff
@@ -1759,3 +1735,56 @@ func selectVRGNamespace(
 		return drpc.Namespace, nil
 	}
 }
+
+func addOrUpdateCondition(conditions *[]metav1.Condition, conditionType string,
+	observedGeneration int64, status metav1.ConditionStatus, reason, msg string,
+) bool {
+	newCondition := metav1.Condition{
+		Type:               conditionType,
+		Status:             status,
+		ObservedGeneration: observedGeneration,
+		LastTransitionTime: metav1.Now(),
+		Reason:             reason,
+		Message:            msg,
+	}
+
+	existingCondition := findCondition(*conditions, conditionType)
+	if existingCondition == nil ||
+		existingCondition.Status != newCondition.Status ||
+		existingCondition.ObservedGeneration != newCondition.ObservedGeneration ||
+		existingCondition.Reason != newCondition.Reason ||
+		existingCondition.Message != newCondition.Message {
```

Comment:
Can be nice to  extract this to conditionsEqual() (like https://pkg.go.dev/bytes#Equal) later.